packages/tizen_rpc_port/tizen/src/rpc_port_proxy.cc

```diff
@@ -0,0 +1,223 @@
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_connect(handle_, appid_.c_str(), port_name_.c_str());
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  connected_ = true;
+  return CreateResult(RPC_PORT_ERROR_NONE);
+}
+
+RpcPortResult RpcPortProxy::ProxyConnectSync(EventSink sink) {
+  LOG_DEBUG("ProxyConnectSync: %s/%s", appid_.c_str(), port_name_.c_str());
+  if (connected_) {
+    LOG_ERROR("Already connected");
+    return CreateResult(RPC_PORT_ERROR_INVALID_PARAMETER);
+  }
+
+  event_sink_ = std::move(sink);
+  int ret =
+      rpc_port_proxy_add_connected_event_cb(handle_, OnConnectedEvent, this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_add_disconnected_event_cb(handle_, OnDisconnectedEvent,
+                                                 this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_add_rejected_event_cb(handle_, OnRejectedEvent, this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_add_received_event_cb(handle_, OnReceivedEvent, this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret =
+      rpc_port_proxy_connect_sync(handle_, appid_.c_str(), port_name_.c_str());
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  connected_ = true;
+  return CreateResult(RPC_PORT_ERROR_NONE);
+}
+
+RpcPortResult RpcPortProxy::GetPort(int32_t type, RpcPort** port) {
+  rpc_port_h port_native = nullptr;
+  int ret = rpc_port_proxy_get_port(
+      handle_, static_cast<rpc_port_port_type_e>(type), &port_native);
+  if (ret != RPC_PORT_ERROR_NONE) return CreateResult(ret);
+
+  *port = new RpcPort(port_native, type);
+  return CreateResult(ret);
+}
+
+void RpcPortProxy::OnConnectedEvent(const char* receiver, const char* port_name,
+                                    rpc_port_h port, void* data) {
+  LOG_DEBUG("OnConnectedEvent, receiver(%s), port_name(%s)", receiver,
+            port_name);
+  if (receiver == nullptr || port_name == nullptr || port == nullptr) {
+    LOG_ERROR("Invalid parameter");
+    return;
+  }
+
+  auto* proxy = static_cast<RpcPortProxy*>(data);
+  auto map = CreateEncodableMap("connected", receiver, port_name);
+  proxy->event_sink_->Success(flutter::EncodableValue(map));
+}
+
+void RpcPortProxy::OnDisconnectedEvent(const char* receiver,
+                                       const char* port_name, void* data) {
+  LOG_DEBUG("OnDisconnectedEvent, receiver(%s), port_name(%s)", receiver,
+            port_name);
+  if (receiver == nullptr || port_name == nullptr || data == nullptr) {
+    LOG_ERROR("Invalid parameter");
+    return;
+  }
+
+  auto* proxy = static_cast<RpcPortProxy*>(data);
+  auto map = CreateEncodableMap("disconnected", receiver, port_name);
```

Comment:
Need to update connected state here.
```suggestion
  auto* proxy = static_cast<RpcPortProxy*>(data);
  proxy->connected_ = false;
  auto map = CreateEncodableMap("disconnected", receiver, port_name);
```

Reply:
I removed connected_.
It seems unnecessary.